tests/installation/ipxe_install.pm

```diff
@@ -47,12 +47,12 @@ sub poweron_host {
 
 sub set_pxe_boot {
     while (1) {
-        my $stdout = ipmitool('chassis bootparam get 5');
-        last if $stdout =~ m/Boot Flag Valid.*Force PXE/s;
         diag "setting boot device to pxe";
         my $options = get_var('IPXE_UEFI') ? 'options=efiboot' : '';
         ipmitool("chassis bootdev pxe ${options}");
         sleep(3);
+        my $stdout = ipmitool('chassis bootparam get 5');
+        last if $stdout =~ m/Force PXE/s;
```

Comment:
Why do you need to get rid of ```Boot Flag Valid``` ? There is no ```Boot Flag Valid``` ??? or ```Boot Flag Invalid``` ???

Reply:
It is NOT always accountable. Take vh015 for example. It is always invalid, although it works.

```
        vh081:~ # ipmitool -H 10.67.135.15 -U root -P 'xx' -I lanplus chassis bootdev disk
        Set Boot Device to disk
        vh081:~ # ipmitool -H 10.67.135.15  -U root -P 'xx' -I lanplus chassis bootparam get 5
        Boot parameter version: 1
        Boot parameter 5 is valid/unlocked
        Boot parameter data: 0008000000
        Boot Flags :
        - Boot Flag Invalid
        - Options apply to only next boot
        - BIOS PC Compatible (legacy) boot
        - Boot Device Selector : Force Boot from default Hard-Drive
        - BIOS verbosity : System Default
        - Console Redirection control : Console redirection occurs per BIOS configuration setting (default)
        - BIOS Mux Control Override : BIOS uses recommended setting of the mux at the end of POST
```

Reply:
But I think this validation here is always after 
```
ipmitool -H 10.67.135.15 -U root -P 'xx' -I lanplus chassis bootdev pxe
```
So you do not need to make this change at all. It always boot from ipxe in the first place.

Reply:
It is more guaranteed to have some check for pxe boot. Let alone the sense of while(1) loop here to ensure that.

Reply:
It is more guaranteed to have some check for pxe boot. Let alone the sense of while(1) loop here to ensure that.

Reply:
This change does not make any sense to me. Your usb boot actually does not affect this part.